### DISCUSSION TITLE
Find everything that DOES NOT MATCH a specific string

### DISCUSSION THREAD

**Discussion by jwankhalaf-cx** [BurntSushi/ripgrep#2494] [Q&A] [answered]
Hello everyone

I am working in .NET codebase and I want to find all namespaces that do not star with `namespace ProjectX.`. Is this possible with `ripgrep`?

For example, say I have a 100 C# code files, and most have their namespace declaration as:

`namespace ProjectX.`

But a few had:

`namespace SomethingElse.`

I want to ripgrep to list those that **do not** have the namespace `namespace ProjectX.`

How do I go about achieving that?

**Comment by s-p-turner** [comment]
There may be a more efficient way, but something like this might work:

rg '^namespace[[:space:]]+' . | rg -v 'ProjectX'

**Reply by BurntSushi** [reply]
Hah, beat me to it! Nice.

**Comment by BurntSushi** [comment, marked as answer]
I would probably do something like `rg '^namespace ' | rg -v ProjectX` and see how that works.

**Reply by jwankhalaf-cx** [reply]
Thank you!
